### DISCUSSION TITLE
pad_left/right units for GPL and repair_clock_inverters description

### DISCUSSION THREAD

**Discussion by lustefan97** [The-OpenROAD-Project/OpenROAD#4495] [Q&A] [answered]
I looked in OpenROAD documentation for Global Placement and Gate Resizer to get more detail about the units of the pad_left/right parameters of the `global_placement` command [(GPL)](https://github.com/The-OpenROAD-Project/OpenROAD/tree/master/src/gpl) and the use/behavior of `repair_clock_inverters` [(RSZ).](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/rsz/README.md) 

Both the variables units description is missing (while present for DPL (site count units), but since the two tools are different I don't want to assume it is the same for both) in GPL, and there is no description concerning `repair_clock_inverters` use case or behavior description (like there would be for `repair_timing` or `repair_design`.

What are the units of pad_left/right, and what does `repair_clock_inverters` do/how to use it ?

**Comment by maliberty** [comment, marked as answer]
The units are sites from the LEF.

repair_clock_inverters is useful when you have an inverter in the clock tree that would split the tree.  It clones the inverter so you have one per flop instead.

**Reply by lustefan97** [reply]
When should repair_clock_inverters be used in a script ? After creation of the CT and after repair_clock_nets ? Or it could be used before CTS with little to no repercussion ?

**Comment by maliberty** [comment]
before CTS see https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/181e9133776117ea1b9f74dbacbfdaadff8c331b/flow/scripts/cts.tcl#L7

You might benefit from investigating how ORFS works

**Reply by lustefan97** [reply]
@maliberty Adding a description to the command documentation in the user guide could also help... I already used it before clock CTS without knowing why but look into the Gate Resizer to know why it is so, I couldn't find anything in the "Read the Docs"

**Reply by maliberty** [reply]
https://github.com/The-OpenROAD-Project/OpenROAD/pull/4499
